pkg/pillar/cmd/volumemgr/initialvolumestatus.go

```diff
@@ -141,3 +141,34 @@ func deleteFile(filelocation string) {
 			filelocation, err.Error())
 	}
 }
+
+// gcPendingCreateVolume remove volumes not created on previous boot
+func gcPendingCreateVolume(ctx *volumemgrContext) {
+	log.Trace("gcPendingCreateVolume")
+
+	for _, obj := range ctx.pubVolumeCreatePending.GetAll() {
+		vcp := obj.(types.VolumeCreatePending)
+		if vcp.IsContainer() {
+			if err := ctx.casClient.RemoveContainerRootDir(vcp.PathName()); err != nil {
+				log.Errorf("gcPendingCreateVolume: error removing container root dir: %s", err)
```

Comment:
In this and the two other error cases below, is it still reasonable to unpublish the VolumeCreatePending?
In these cases can't there still be remnants of the directory/file/zvol?

Reply:
I put additional checks inside the function to not go into deletion in case of no presence of files/directories. And in case of errors now we do not unpublish pending object.